include/arch/xtensa/log.h

```diff
@@ -0,0 +1,15 @@
+/*
```

Comment:
Are all these just for printing exception messages? Or do you intend these to be used for all log messages? Because you define these in `include/arch/xtensa/log.h` which means these should be used for all logging in Xtensa.

Reply:
These changes are for all log messages in `xtensa-asm2.c` and `fatal.c`. These changes are required to handle [Cache disabled but cache memory region accessed](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/fatal-errors.html#cache-disabled-but-cached-memory-region-accessed) exception.

Using LOG module in Zephyr will lead to flash cache access and cause another exception from panic handler.

Reply:
The problem with overriding `LOG_*()` is that the formatted output would not be consistent, since all the formatting code from the logging subsystem is bypassed. Also, by overriding `LOG_*()`, you no longer have the option to send the log message through, for example, the network logging backend. If remote logging is required, the exception messages (fatal and non-fatal ones) are no longer being sent remotely through some backends.

Reply:
* While flash operations like write, erase, etc. are performed on ESP32, flash cache and hence XIP is disabled. So, the code present in flash cache cannot be accessed. When a crash occurs in such scenario, code from flash cannot be accessed. That's why entire panic handler and all the dependent functions must be placed in internal RAM.
* LOG module is one of the dependencies of panic handler. Also, LOG module has a lot of dependencies. Hence, it is not possible to move it to internal RAM as it will eat up memory required for end application.
* In ESP-IDF, minimalistic printf implementation is used in panic handler which directly calls UART APIs and is redirected to ROM code. This is preferred because normal printf implementation can allocate dynamic memory and it is not recommended to allocate memory when system is crashed.
* As ESP32 support is in baseline mode currently, it will be very difficult to debug potential exceptions without these changes. Also, we are providing a configurable option to enable/disable backtrace feature. This config option can be enabled in debugging scenarios only.

@dcpleung Please let me know your views on this.

Reply:
please don't hack the definitions of LOG like this.

Reply:
Have moved log module to internal memory which fixes the issue. This change is removed.